internal/migration/precheck.go

```diff
@@ -49,6 +48,10 @@ func SourcePrecheck(
 		return errors.Trace(err)
 	}
 
+	if err := statusService.CheckUnitStatusesReadyForMigration(ctx); err != nil {
```

Comment:
I may be wrong, but it seems you were not previously checking for unit presence, and now you are. Is this change on purpose? (Its good that you are now.)